Build error on Ubuntu 12

I have trouble building usrsctp-neat on Ubuntu 12 on Travis CI. This is the output:

> CFLAGS=-fPIC cmake -DCMAKE_INSTALL_PREFIX=/home/travis/build/rawrtc/rawrtc/build/prefix -DSCTP_DEBUG=1 ..
> make install -j12
> Scanning dependencies of target usrsctp-static
> Scanning dependencies of target usrsctp
> [  2%] [  2%] [  4%] [  5%] [  7%] Building C object usrsctplib/CMakeFiles/usrsctp.dir/netinet6/sctp6_usrreq.c.o
> [  8%] [ 10%] Building C object usrsctplib/CMakeFiles/usrsctp.dir/netinet/sctp_callout.c.o
> [ 11%] [ 13%] Building C object usrsctplib/CMakeFiles/usrsctp.dir/netinet/sctp_input.c.o
> [ 16%] [ 16%] [ 17%] Building C object usrsctplib/CMakeFiles/usrsctp-static.dir/netinet6/sctp6_usrreq.c.o
> Building C object usrsctplib/CMakeFiles/usrsctp.dir/netinet/sctp_sha1.c.o
> Building C object usrsctplib/CMakeFiles/usrsctp.dir/netinet/sctp_userspace.c.o
> Building C object usrsctplib/CMakeFiles/usrsctp-static.dir/netinet/sctp_input.c.o
> Building C object usrsctplib/CMakeFiles/usrsctp-static.dir/netinet/sctp_callout.c.o
> Building C object usrsctplib/CMakeFiles/usrsctp.dir/user_mbuf.c.o
> Building C object usrsctplib/CMakeFiles/usrsctp-static.dir/netinet/sctp_sha1.c.o
> Building C object usrsctplib/CMakeFiles/usrsctp-static.dir/netinet/sctp_userspace.c.o
> Building C object usrsctplib/CMakeFiles/usrsctp.dir/netinet/sctp_asconf.c.o
> [ 19%] Building C object usrsctplib/CMakeFiles/usrsctp-static.dir/user_mbuf.c.o
> [ 20%] Building C object usrsctplib/CMakeFiles/usrsctp.dir/netinet/sctp_cc_functions.c.o
> [ 22%] [ 23%] Building C object usrsctplib/CMakeFiles/usrsctp-static.dir/netinet/sctp_asconf.c.o
> [ 25%] Building C object usrsctplib/CMakeFiles/usrsctp-static.dir/netinet/sctp_cc_functions.c.o
> [ 26%] Building C object usrsctplib/CMakeFiles/usrsctp-static.dir/netinet/sctp_output.c.o
> Building C object usrsctplib/CMakeFiles/usrsctp.dir/netinet/sctp_output.c.o
> [ 28%] Building C object usrsctplib/CMakeFiles/usrsctp-static.dir/netinet/sctp_ss_functions.c.o
> [ 29%] Building C object usrsctplib/CMakeFiles/usrsctp-static.dir/netinet/sctp_usrreq.c.o
> [ 31%] [ 32%] Building C object usrsctplib/CMakeFiles/usrsctp-static.dir/user_recv_thread.c.o
> Building C object usrsctplib/CMakeFiles/usrsctp-static.dir/netinet/sctp_auth.c.o
> [ 34%] Building C object usrsctplib/CMakeFiles/usrsctp-static.dir/netinet/sctp_crc32.c.o
> /home/travis/build/rawrtc/rawrtc/build/dependencies/usrsctp/usrsctplib/user_recv_thread.c: In function ‘recv_icmp’:
> /home/travis/build/rawrtc/rawrtc/build/dependencies/usrsctp/usrsctplib/user_recv_thread.c:408:29: error: ‘struct udphdr’ has no member named ‘uh_sport’
> /home/travis/build/rawrtc/rawrtc/build/dependencies/usrsctp/usrsctplib/user_recv_thread.c: In function ‘recv_icmp6’:
> /home/travis/build/rawrtc/rawrtc/build/dependencies/usrsctp/usrsctplib/user_recv_thread.c:845:29: error: ‘struct udphdr’ has no member named ‘uh_sport’
> make[2]: *** [usrsctplib/CMakeFiles/usrsctp-static.dir/user_recv_thread.c.o] Error 1
> make[2]: *** Waiting for unfinished jobs....
> [ 35%] Building C object usrsctplib/CMakeFiles/usrsctp.dir/netinet/sctp_ss_functions.c.o
> [ 37%] Building C object usrsctplib/CMakeFiles/usrsctp.dir/netinet/sctp_usrreq.c.o
> [ 38%] Building C object usrsctplib/CMakeFiles/usrsctp.dir/user_recv_thread.c.o
> [ 40%] Building C object usrsctplib/CMakeFiles/usrsctp.dir/netinet/sctp_auth.c.o
> [ 41%] Building C object usrsctplib/CMakeFiles/usrsctp.dir/netinet/sctp_crc32.c.o
> /home/travis/build/rawrtc/rawrtc/build/dependencies/usrsctp/usrsctplib/user_recv_thread.c: In function ‘recv_icmp’:
> /home/travis/build/rawrtc/rawrtc/build/dependencies/usrsctp/usrsctplib/user_recv_thread.c:408:29: error: ‘struct udphdr’ has no member named ‘uh_sport’
> /home/travis/build/rawrtc/rawrtc/build/dependencies/usrsctp/usrsctplib/user_recv_thread.c: In function ‘recv_icmp6’:
> /home/travis/build/rawrtc/rawrtc/build/dependencies/usrsctp/usrsctplib/user_recv_thread.c:845:29: error: ‘struct udphdr’ has no member named ‘uh_sport’
> make[2]: *** [usrsctplib/CMakeFiles/usrsctp.dir/user_recv_thread.c.o] Error 1
> make[2]: *** Waiting for unfinished jobs....
> make[1]: *** [usrsctplib/CMakeFiles/usrsctp-static.dir/all] Error 2
> make[1]: *** Waiting for unfinished jobs....
> make[1]: *** [usrsctplib/CMakeFiles/usrsctp.dir/all] Error 2
> make: *** [all] Error 2

You can find the full log [here](https://api.travis-ci.org/jobs/212465030/log.txt?deansi=true) which also includes precise information about the build system.

--- a/usrsctplib/netinet/sctputil.c
+++ b/usrsctplib/netinet/sctputil.c
@@ -53,6 +53,9 @@ __FBSDID("$FreeBSD: head/sys/netinet/sctputil.c 310590 2016-12-26 11:06:41Z tuex
 #include <netinet/sctp_auth.h>
 #include <netinet/sctp_asconf.h>
 #include <netinet/sctp_bsd_addr.h>
+#if defined(__Userspace_os_Linux)
+#define __FAVOR_BSD    /* (on Ubuntu at least) enables UDP header field names like BSD in RFC 768 */
+#endif
 #if defined(__Userspace__)
 #include <netinet/sctp_constants.h>
 #if !defined(__Userspace_os_Windows)

--- a/usrsctplib/user_recv_thread.c
+++ b/usrsctplib/user_recv_thread.c
@@ -57,6 +57,9 @@
 #endif
 #endif
 #endif
+#if defined(__Userspace_os_Linux)
+#define __FAVOR_BSD    /* (on Ubuntu at least) enables UDP header field names like BSD in RFC 768 */
+#endif
 #if defined(__Userspace__)
 #ifdef INET6
 #include <netinet6/sctp6_var.h>

